compiler/z/codegen/OMRRegister.cpp

```diff
@@ -70,8 +70,12 @@ void
 OMR::Z::Register::setPlaceholderReg()
    {
 #if defined(TR_TARGET_64BIT)
-   self()->setIs64BitReg(true);
+   if (self()->getKind() == TR_GPR)
+      {
+      self()->setIs64BitReg(true);
+      }
```

Comment:
We added an assert to make sure the `is64BitReg()` query is only used for GPRs, however we do have placeholder FPR registers so we avoid setting this flag unless it's a GPR.